src/main/java/sirius/kernel/xml/XMLReader.java

```diff
@@ -149,6 +149,14 @@ public void parse(InputStream stream) throws IOException {
         private static final long serialVersionUID = -7454219131982518216L;
     }
 
+    /**
```

Comment:
do we really need 2 distinct exceptions for the exact same purpose?

why not wrap processing in its own task context and then simply call taskContext.cancel();